go/vt/sqlparser/sql.y

```diff
@@ -354,10 +391,449 @@ set_statement:
+  {
+    $$ = BoolVal(false)
+  }
+| AUTO_INCREMENT
+  {
+    $$ = BoolVal(true)
+  }
+
+charset_opt:
+  {
+    $$ = ""
+  }
+| CHARACTER SET ID
+  {
+    $$ = string($3)
+  }
+| CHARACTER SET BINARY
+  {
+    $$ = string($3)
+  }
+
+collate_opt:
+  {
+    $$ = ""
+  }
+| COLLATE ID
+  {
+    $$ = string($2)
+  }
+
+column_key_opt:
+  {
+    $$ = ColKeyNone
+  }
+| PRIMARY KEY
+  {
+    $$ = ColKeyPrimary
+  }
+| KEY
+  {
+    $$ = ColKey
+  }
+| UNIQUE KEY
+  {
+    $$ = ColKeyUnique
+  }
+| UNIQUE
+  {
+    $$ = ColKeyUnique
+  }
+
+column_comment_opt:
+  {
+    $$ = nil
+  }
+| COMMENT_KEYWORD STRING
+  {
+    $$ = NewStrVal($2)
+  }
+
+enum_values:
+  STRING
+  {
+    $$ = make([]string, 0, 4)
+    $$ = append($$, "'" + string($1) + "'")
+  }
+| enum_values ',' STRING
+  {
+    $$ = append($1, "'" + string($3) + "'")
+  }
+
+column_definition:
+  ID column_type null_opt column_default_opt auto_increment_opt column_key_opt column_comment_opt
+  {
+    $2.NotNull = $3
+    $2.Default = $4
+    $2.Autoincrement = $5
+    $2.KeyOpt = $6
+    $2.Comment = $7
+    $$ = &ColumnDefinition{Name: NewColIdent(string($1)), Type: $2}
+  }
+
+index_definition:
+  index_info '(' index_column_list ')'
+  {
+    $$ = &IndexDefinition{Info: $1, Columns: $3}
+  }
+
+index_info:
+  PRIMARY KEY
+  {
+    $$ = &IndexInfo{Primary: true, Name: NewColIdent("PRIMARY"), Unique: true}
+  }
+| UNIQUE KEY ID
+  {
+    $$ = &IndexInfo{Name: NewColIdent(string($3)), Unique: true}
+  }
+| KEY ID
+  {
+    $$ = &IndexInfo{Name: NewColIdent(string($2)), Unique: false}
+  }
+
+index_column_list:
+  index_column
+  {
+    $$ = []*IndexColumn{$1}
+  }
+| index_column_list ',' index_column
+  {
+    $$ = append($$, $3)
+  }
+
+index_column:
+  sql_id length_opt
+  {
+      $$ = &IndexColumn{Column: $1, Length: $2}
+  }
+
+table_spec:
+  '(' table_column_list ')' table_option_list
+  {
+    $$ = $2
+    $$.Options = $4
+  }
+
+table_column_list:
+  column_definition
+  {
+    $$ = &TableSpec{}
+    $$.AddColumn($1)
+  }
+| table_column_list ',' column_definition
+  {
+    $$.AddColumn($3)
+  }
+| table_column_list ',' index_definition
+  {
+    $$.AddIndex($3)
+  }
+
+table_option_list:
+  {
+    $$ = ""
+  }
+| table_option
+  {
+    $$ = " " + string($1)
+  }
+| table_option_list ',' table_option
+  {
+    $$ = string($1) + ", " + string($3)
+  }
+
+// rather than explicitly parsing the various keywords for table options,
+// just accept any number of keywords, IDs, strings, numbers, and '='
+table_option:
+  table_opt_value
+  {
+    $$ = $1
+  }
+| table_option table_opt_value
+  {
+    $$ = $1 + " " + $2
+  }
+| table_option '=' table_opt_value
+  {
+    $$ = $1 + "=" + $3
+  }
+
+table_opt_value:
+  reserved_sql_id
+  {
+    $$ = $1.String()
+  }
+| STRING
+  {
+    $$ = "'" + string($1) + "'"
+  }
+| INTEGRAL
+  {
+    $$ = string($1)
+  }
+
 create_statement:
-  CREATE TABLE not_exists_opt table_name ddl_force_eof
+  CREATE TABLE not_exists_opt table_name table_spec
```

Comment:
Can you try this construct:
```
func setDDL(yylex interface{}, ddl *DDL) {
  yylex.(*Tokenizer).ddl = ddl  // add ddl as member of Tokenizer
}
...


create_statement:
  create_prefix tablet_spec
  {
    $0.TableSpec = $2
  }

create_prefix:
  CREATE TABLE not_exists_opt table_name
  {
    $$ = &DDL{...}
    setDDL(yylex, $$)
  }
```

If you did this, we should still be able to return a successful DDL object if the `CREATE` fails to parse.